### PR TITLE
Support SpaceDepth ops in the CUDA and ROCM EPs

### PR DESCRIPTION
**Pull request by hariharans29**
**Description**: As title

**Motivation and Context**
Improve coverage of ops on GPU
https://github.com/microsoft/onnxruntime/issues/7681

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/space_depth_ops.cc [hunk 1]
```diff
@@ -16,103 +16,132 @@ ONNX_CPU_OPERATOR_VERSIONED_KERNEL(
     SpaceToDepth,
     1,
     12,
-    KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
-    SpaceToDepth<float>);
+    KernelDefBuilder()
+        .TypeConstraint("T", {DataTypeImpl::GetTensorType<float>(),
+                              DataTypeImpl::GetTensorType<double>()}),
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/tensor/space_depth_ops.cc:21]: can support double - I think it has more advantages than dis-advantages. The binary size increase should be very marginal as the templated code section is very small.

File: tools/ci_build/amd_hipify.py [hunk 2]
```diff
@@ -144,6 +144,8 @@
                 'tensor/resize_impl.h',
                 'tensor/transpose.cc',
                 'tensor/transpose.h',
+                'tensor/space_depth_ops.cc',
+                'tensor/space_depth_ops.h',
```
Comment (weixingzhang) [inline tools/ci_build/amd_hipify.py:148]: Can you add the code below to this file line #227? Then the kernels for ROCm can be generated from CUDA implementation.

s = s.replace('cublasHandle', 'RocblasHandle')
s = s.replace('cublas_handle', 'rocblas_handle')
s = s.replace('cublasHandle_t', 'rocblas_handle')

Comment (hariharans29) [reply tools/ci_build/amd_hipify.py:148]: Ok. So, we can remove the ROCM implementation files all together ?

Comment (weixingzhang) [reply tools/ci_build/amd_hipify.py:148]: yes.

Comment (hariharans29) [reply tools/ci_build/amd_hipify.py]: Done

**Review by weixingzhang — APPROVED** [review]

**Pull request merged**